File: FrontEnd/cms/src/services/StudentService.js
import axios from 'axios';

const STUDENT_API_BASE_URL="http://localhost:8080/api/v1/students";

class StudentService
{
    getAllStudents()
    {
        return axios.get(STUDENT_API_BASE_URL + "/listOfStudents");
    }

    addStudent(student)
    {
        return axios.post(STUDENT_API_BASE_URL,student);
    }

    getStudentById(studentId)
    {
        return axios.get(STUDENT_API_BASE_URL+ "/singleStudent/"+ studentId);
    }

    updateStudent(studentId,student)
    {
        return axios.put(STUDENT_API_BASE_URL+ "/updateStudent/"+studentId,student);
    }

    deleteStudent(studentId)
    {
        return axios.delete(STUDENT_API_BASE_URL + "/deleteSingleStudent/"+studentId);
    }

}

export default new StudentService();

File: BackEnd/Backend/CollegeManagementSystem/src/main/java/com/collegemanagementsystem/service/StudentService.java
package com.collegemanagementsystem.service;

import java.util.List;

import com.collegemanagementsystem.dto.StudentDto;

public interface StudentService {
	String addStudent(StudentDto studentDto);
	StudentDto getStudentById(Long id);
	List<StudentDto> getAllStudents();
	String updateStudent(StudentDto studentDto,Long id);
	String deleteStudentById(Long id);
}
